server/TypeDBServer.java

```diff
@@ -109,14 +111,28 @@ protected TypeDBServer(CoreConfig config, boolean debug, Factory factory) {
         server = rpcServer();
         Thread.setDefaultUncaughtExceptionHandler(
                 (t, e) -> {
-                    logger().error(UNCAUGHT_EXCEPTION.message(t.getName(), e), e);
-                    close();
-                    System.exit(1);
+                    try {
+                        logger().error(UNCAUGHT_ERROR.message(t.getName(), e), e);
+                        close();
+                        System.exit(1);
+                    } catch (TypeDBCheckedException ex) {
+                        logger().error("Failed to shut down cleanly, performing hard stop.");
+                        Runtime.getRuntime().halt(1);
+                    } catch (Throwable error) {
+                        // another thread will do the close
+                    }
                 }
         );
         Runtime.getRuntime().addShutdownHook(
-                NamedThreadFactory.create(TypeDBServer.class, "shutdown").newThread(this::close)
+                NamedThreadFactory.create(TypeDBServer.class, "shutdown").newThread(() -> {
+                    try {
+                        close();
+                    } catch (Throwable error) {
+                        logger().error("Error during shutdown: ", error);
+                    }
```

Comment:
If the user hits ctrl+c, we want to initiate an orderly shutdown. However, if this fails for any reason we log the error before returning control to the JVM shutdown sequence